beacon_node/network/src/network_beacon_processor/sync_methods.rs

```diff
@@ -44,6 +45,60 @@ struct ChainSegmentFailed {
     peer_action: Option<PeerAction>,
 }
 
+#[derive(Debug)]
+pub enum LookupSyncProcessingResult {
+    FullyImported,
+    ImportedMissingComponents,
+    Error(ErrorCategory),
+    ParentUnknown(Hash256),
+    Ignored,
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum ErrorCategory {
+    /// Internal Errors (not caused by peers).
+    ///
+    /// An internal no-retry error is permanent and block processing should not be
+    /// re-attempted.
+    Internal { retry: bool },
+    /// Errors caused by faulty / malicious peers.
+    ///
+    /// No retry errors are deterministic against the block's root. Re-downloading data
+    /// key-ed by block root MUST result in the same no-retry error (i.e. invalid parent,
+    /// invalid state root, etc).
+    ///
+    /// The error also indicates which block component index is malicious if applicable.
+    Malicious { retry: bool, index: usize },
+}
+
+impl From<ErrorCategory> for LookupSyncProcessingResult {
+    fn from(e: ErrorCategory) -> Self {
+        Self::Error(e)
+    }
+}
+
+impl ErrorCategory {
+    // Helper functions for readibility on large match statements
+    pub fn internal_no_retry() -> Self {
+        Self::Internal { retry: false }
+    }
+    pub fn internal_retry() -> Self {
+        Self::Internal { retry: true }
+    }
+    pub fn malicious_no_retry() -> Self {
+        Self::Malicious {
+            retry: false,
+            index: 0,
```

Comment:
I think it make sense to make this `Option`, otherwise we can't distinguish between `0` index or `None`